src/Eleventy.js

```diff
@@ -33,7 +39,13 @@ const eventBus = require("./EventBus");
  * @param {String} output - Where to write rendered files to.
  * @returns {module:11ty/eleventy/Eleventy~Eleventy}
  */
-class Eleventy {
+export default class Eleventy {
+  static async from(input, output, options = {}, eleventyConfig = null) {
```

Comment:
fwiw you can make a constructor async with this pattern:

```js
export default class SomeClass {
  constructor() {
    return (async () => {
      // do something async...
      return this;
    })();
  }
}
const someObject = await new SomeClass();
```

It is unfortunately not compatible with TypeScript, though: https://github.com/Microsoft/TypeScript/issues/29138

Reply:
very cool. nice pattern, and in this PR i think .from is better because it opens the door for future TS support. thanks for the tip, very cool